vllm/v1/worker/gpu_model_runner.py

```diff
@@ -632,8 +632,36 @@ def _update_states(self, scheduler_output: "SchedulerOutput") -> None:
             num_computed_tokens = req_data.num_computed_tokens[i]
             new_block_ids = req_data.new_block_ids[i]
             resumed_from_preemption = req_data.resumed_from_preemption[i]
+            num_output_tokens = req_data.num_output_tokens[i]
 
             # Update the cached states.
+
+            if (num_computed_tokens <= req_state.num_computed_tokens and
+                    num_output_tokens == req_state.rollback_num_output_tokens):
+                # Request was rescheduled after a sync-KV-load failure. Remove
+                # any tokens sampled past the rollback checkpoint and restore
```

Comment:
Would it be better / might it be possible to make indication of this case more explicit? It seems `rollback_num_output_tokens` is only used in an indirect way anyhow (comparing with `num_output_tokens`).